File: frontend/src/types/checkin.ts
export type CheckInStatus = "not_started" | "on_track" | "completed";

export interface CheckInCreate {
  actual_value?: number | null;
  status: CheckInStatus;
}

export interface ManagerCommentCreate {
  manager_comment: string;
}

export interface CheckInRead {
  id: string;
  goal_id: string;
  quarter: number;
  year: number;
  actual_value?: number | null;
  status: CheckInStatus;
  manager_comment?: string | null;
  updated_by: string;
  created_at: string;
  updated_at: string;
}

export interface CheckInGoalRead {
  goal_id: string;
  goal_sheet_id: string;
  owner_id: string;
  goal_title: string;
  thrust_area: string;
  uom_type: string;
  direction?: string | null;
  target_value?: number | null;
  target_date?: string | null;
  current_actual?: number | null;
  quarter: number;
  year: number;
  checkin_status: CheckInStatus;
  manager_comment?: string | null;
  progress_score?: number | null;
  is_shared: boolean;
  primary_owner_id?: string | null;
}

File: backend/app/schemas/checkin.py
from datetime import date, datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field

CheckInStatus = Literal["not_started", "on_track", "completed"]


class CheckInCreate(BaseModel):
    actual_value: Decimal | None = Field(default=None, ge=0, max_digits=12, decimal_places=2)
    status: CheckInStatus


class ManagerCommentCreate(BaseModel):
    manager_comment: str = Field(min_length=1, max_length=3000)


class CheckInRead(BaseModel):
    id: str
    goal_id: str
    quarter: int
    year: int
    actual_value: Decimal | None
    status: CheckInStatus
    manager_comment: str | None
    updated_by: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class CheckInGoalRead(BaseModel):
    goal_id: str
    goal_sheet_id: str
    owner_id: str
    goal_title: str
    thrust_area: str
    uom_type: str
    direction: str | None
    target_value: Decimal | None
    target_date: date | None
    current_actual: Decimal | None
    quarter: int
    year: int
    checkin_status: CheckInStatus
    manager_comment: str | None
    progress_score: float | None
    is_shared: bool
    primary_owner_id: str | None
